habitat-hitl/habitat_hitl/_internal/config_helper.py

```diff
@@ -19,6 +21,14 @@ def update_config(
     debug_third_person_height=None,
 ):
     with habitat.config.read_write(config):  # type: ignore
+        # Move the metadata info to the habitat simulator config
+        if "metadata" in config:
```

Comment:
Not sure if this repo has an existing solution for metadata.

Reply:
Can you check how are the affordances being read in the base code for this change?

Reply:
The only usages I see of metadata in phase3 are things like 

habitat-lab/habitat/datasets/rearrange/rearrange_generator.py
147:                self.sim.metadata_mediator.get_scene_handles(),
506:            mm=self.sim.metadata_mediator, scene_handle=ep_scene_handle

which is using the habitat_sim metadata, which is not what we need for the object states.

and 
habitat-hitl/habitat_hitl/_internal/networking/keyframe_utils.py
53:    # add or update instance metadata
54:    if "metadata" in inc_keyframe:
55:        ensure_list(consolidated_keyframe, "metadata")
56:        for metadata_container in inc_keyframe["metadata"]:
57:            key = metadata_container["instanceKey"]
58:            metadata = metadata_container["metadata"]
59:            if "metadata" in consolidated_keyframe:

Which appears to be a part of the networking.

I see no usages of "affordances", "affordance", or "afford" anywhere in phase3 in this repo other than usages in the object state machine code itself. Would you mind pointing me to the code you're referring to if I've somehow missed it?

Reply:
Added here: https://github.com/facebookresearch/habitat-lab/pull/2022